lib/plug/cowboy/drainer.ex

```diff
@@ -0,0 +1,106 @@
+defmodule Plug.Cowboy.Drainer do
+  @moduledoc """
+  GenServer to drain cowboy connections at shutdown.
+
+  When starting `Plug.Cowboy` in a supervision tree, it will create a listener that receives
+  requests and creates a connection process to handle that request. During shutdown, a
+  `Plug.Cowboy` process will immediately exit, closing the listener and any open connections
+  that are still being served. In most cases, it is desireable to allow connections to
+  complete before shutting down.
+
+  This module provides a GenServer that during shutdown will close listeners and wait
+  for connections to complete. It should be placed after other supervised processes that
+  handle cowboy connections.
+
+  ## Options
+
+    * `:refs` - A list of refs to drain. `:all` is also supported and will drain all cowboy
+      listeners, including those started by means other than `Plug.Cowboy`.
+
+    * `:id` - The ID for the process.
+      Defaults to `Plug.Cowboy.Drainer`.
+
+    * `:shutdown` - How long to wait for connections to drain.
+      Defaults to 5000ms.
+
+    * `:drain_check_interval` - How frequently to check if a listener's
+      connections have been drained.
+      Defaults to 1000ms.
+
+  ## Examples
+
+      defmodule Hello.Application do
+        use Application
+        def start(_type, _args) do
+          ...
```

Comment:
I don't think we need to use `...` to signal other code here. We can start with the children directly.